downstairs/src/lib.rs

```diff
@@ -1945,10 +1942,31 @@ impl Downstairs {
      * active Upstairs without causing jobs to be incorrectly sent to the
      * newly active Upstairs.
      */
```

Comment:
Done in 79b7847d2d5bf22289b0a93319ff0de1eb127f48

I ended up deleting most of the comment, because this work lock **wasn't** the thing protecting us (as evidenced by the fact that I could delete it!).  It was the top-level lock on the `Arc<Mutex<Downstairs>>` that was preventing multiple tasks for interfering with the work map.